src/slash/database.ts

```diff
@@ -0,0 +1,68 @@
+import { AutocompleteInteraction, CacheType, ChatInputCommandInteraction, SlashCommandStringOption } from "discord.js";
+import { TournamentStatus } from "../database/interface";
+import { ManualParticipant, ManualTournament } from "../database/orm";
+
+export const tournamentOption = new SlashCommandStringOption()
+	.setName("tournament")
+	.setDescription("The name of the tournament to edit.")
+	.setRequired(true)
+	.setAutocomplete(true);
+
+export async function autocompleteTournament(interaction: AutocompleteInteraction<CacheType>): Promise<void> {
+	if (!interaction.inCachedGuild()) {
+		return;
+	}
+	const partialName = interaction.options.getFocused();
+	const owningDiscordServer = interaction.guildId;
+	const tournaments = await ManualTournament.find({
+		where: [
+			{ owningDiscordServer, status: TournamentStatus.IPR },
+			{ owningDiscordServer, status: TournamentStatus.PREPARING }
+		]
+	});
+	// can we do this natively with .find? should match all for blank input
```

Comment:
Yes, add to the where clause. There is a Postgres function called [starts_with](https://pgpedia.info/s/starts_with.html),  but we can also do `LIKE 'token%'` with case sensitivity off

Reply:
Hmm, that's assuming we'd only want to match the start, I was thinking of matching anywhere in the string, e.g. "Yugioh Chalislime Series" should be matched by "Chalislime".

Reply:
Okay, then just `LIKE '%token%;`

Reply:
You can leave it as is, there is some nuance to this including injection, and in all likelihood there won't be many tournaments that satisify IPR or PREPARING to begin with